### PR TITLE
Enhancements: Code refactored and packages cleanup

### PR DESCRIPTION
**Pull request by saikumarrs**
## Description of the change

- Removed unwanted code.
- Re-factored code w.r.t to eslint suggestions.
- Fixed a minor bug in analytics.js.
- Utility methods relocated.

## Type of change
- [x] Bug fix (non-breaking change that fixes an issue)
- [x] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Related issues

> N/A

### Development

- [x] Lint rules pass locally
- [ ] The code changed/added as part of this pull request has been covered with tests
- [ ] All tests related to the changed code pass in development

### Code review 

- [x]  This pull request has a descriptive title and information useful to a reviewer. There may be a screenshot or screencast attached
- [ ] "Ready for review" label attached to the PR and reviewers mentioned in a comment
- [ ] Changes have been reviewed by at least one other engineer
- [ ] Issue from task tracker has a link to this pull request

---
This change is [Reviewable](https://reviewable.io/reviews/rudderlabs/rudder-sdk-js/483)

### CODE REVIEW

File: utils/linker/index.js [hunk 1]
```diff
@@ -13,31 +13,14 @@ const CHECKSUM_OFFSET_MAX_MIN = 1;
...
-  const linkerObj = parseLinkerParamValue(value);
-  if (!linkerObj) {
-    return null;
-  }
-  const { checksum, serializedIds } = linkerObj;
-  if (!isCheckSumValid(serializedIds, checksum)) {
-    return null;
-  }
-  return deserialize(serializedIds);
-}
```
Comment (MoumitaM) [inline utils/linker/index.js:31]: This `parseLinker` fn is being used in **analytics.js**
In `setAnonymousId` fn.

Comment (saikumarrs) [reply utils/linker/index.js:31]: We haven't deleted any code. The methods have been ordered as per their declaration and usage.

Comment (MoumitaM) [reply utils/linker/index.js:31]: My bad. Missed the below declarations.

**Review by MoumitaM — DISMISSED** [review]
LGTM

File: utils/utils.js [hunk 2]
```diff
@@ -693,18 +693,15 @@ const getSDKUrlInfo = () => {
...
-      sdkURL = curScriptSrc;
-      if (curScriptSrc.endsWith("rudder-analytics-staging.min.js")) {
-        isStaging = true;
+    if (curScriptSrc) {
+      const urlMatches = curScriptSrc.match(
+        /^(https?:)?\/\/.*rudder-analytics(-staging)?(\.min)?\.js$/,
+      );
+      if (urlMatches) {
+        sdkURL = curScriptSrc;
+        isStaging = urlMatches[1] !== undefined;
```
Comment (MoumitaM) [inline utils/utils.js:702]:
```suggestion
        isStaging = urlMatches[2] !== undefined;
```

**Review by MoumitaM — CHANGES_REQUESTED** [review]

File: utils/utils.js (hunk 2 above)
Comment (saikumarrs) [reply utils/utils.js]: Great find. Fixed it.

**Review by MoumitaM — DISMISSED** [review]

**Review by utsabc — APPROVED** [review]

**Comment by saikumarrs** [comment]
Closing this PR to handle these changes later.

**Pull request closed without merging**